### ISSUE TITLE
[BUG/Help] MacOS M1在mps情况下报错， error: input types 'tensor<1x4x1xf16>' and 'tensor<1xf32>' are not broadcast compatible LLVM ERROR: Failed to infer result type(s).

### ISSUE THREAD

**Issue by dydwgmcnl4241** [THUDM/ChatGLM-6B#512]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

在.half().to('mps')时得到正常返回内容。

### Steps To Reproduce

from transformers import AutoTokenizer, AutoModel
tokenizer = AutoTokenizer.from_pretrained("./chatglm-6b", trust_remote_code=True)
model = AutoModel.from_pretrained("./chatglm-6b", trust_remote_code=True).half().to('mps')
model = model.eval()
执行以上代码均正常。

执行到这行代码时报错：
response, history = model.chat(tokenizer, "你好", history=[])

/Users/wilson/miniforge3/lib/python3.10/site-packages/transformers/generation/utils.py:686: UserWarning: MPS: no support for int64 repeats mask, casting it to int32 (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/native/mps/operations/Repeat.mm:236.)
  input_ids = input_ids.repeat_interleave(expand_size, dim=0)
The dtype of attention mask (torch.int64) is not bool
loc("varianceEps"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/97f6331a-ba75-11ed-a4bc-863efbbaf80d/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<1x4x1xf16>' and 'tensor<1xf32>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).

另外如果model用cup的float()方法，运行正常，可以获取到回复内容。

### Environment

```markdown
- OS:MacOS 13.3.1
- Python: 3.10.6
- Transformers:4.27.1
- PyTorch:torch_nightly_env
```

**Comment by dydwgmcnl4241** [comment]
.float().to('mps')
